### ISSUE TITLE
php7.0.2 and phpseclib Deprecated Functionality: Methods with the same name as their class will not be constructors in a future version of PHP; Crypt_Base has a deprecated constructor in ...\vendor\phpseclib\phpseclib\phpseclib\Crypt\Base.php on line 121

### ISSUE THREAD

**Issue by csmarvz** [magento/magento2#3063]
It seems that the version of phpseclib (~0.3) is too old for php 7 and doesn't use a constructor _constructor()

**Comment by andimov** [comment]
@csmarvz  thank you for reporting.
Please fill description according to the template and specify magento version.

**Comment by andimov** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
Please feel free to reopen if it's needed.

**Comment by AntonEvers** [comment]
Fixed in https://github.com/magento/magento2/commit/1383a49e742a811a40ead99c515df6fc4ce60e28

**Comment by Ctucker9233** [comment]
@ajpevers for those of us that don't/can't use composer, how do we get the fix?

**Comment by AntonEvers** [comment]
@Ctucker9233 in your "require-dev" section of composer.json, include this dependency:

``` json
{
  "require-dev": {
    "phpseclib/phpseclib": "2.0.4 as 0.3.10"
  }
}
```

And run `composer update`

This will alias 2.0.4 as version 0.3.10 and install it without conflicting with other modules' dependencies.

Needless to say this will install a higher version of phpseclib than required by some modules which means they may be incompatible. Probably it will work fine, but I have no guarantees for you.

**Comment by Ctucker9233** [comment]
@ajpevers I'll leave it alone for now. As I said, I don't have access to composer, so it looks like there's no way for me to get the fix right now. Any idea which version of magento will have it?

**Comment by AntonEvers** [comment]
@Ctucker9233 It is fixed since 2.1.0-rc1 so all the way up to 2.1.2 you will have phpseclib/phpseclib 2.0.*

**Comment by Ctucker9233** [comment]
@ajpevers Thanks
